pkg/monitor/agent/agent.go

```diff
@@ -373,6 +382,7 @@ func (a *Agent) processPerfRecord(scopedLog *logrus.Entry, record perf.Record) {
 		})
 
 	} else {
+		a.limiter.Wait(a.ctx)
```

Comment:
Wouldn't it be better to wait before reading from the perf event buffer, i.e. at line 347 before the call to `monitorEvents.Read()`? As is, we'd have a Read -> Wait -> Encode -> Read ... sequence; whereas I'm proposing a Read -> Encode -> Wait -> Read ... sequence.

Reply:
changed